tapdb/assets_store.go

```diff
@@ -1238,6 +1238,37 @@ func (a *AssetStore) FetchProof(ctx context.Context,
 	return diskProof, nil
 }
 
+// HasProof returns true if the proof for the given locator exists. This is
+// intended to be a performance optimized lookup compared to fetching a proof
+// and checking for ErrProofNotFound.
+func (a *AssetStore) HasProof(ctx context.Context, locator proof.Locator) (bool,
+	error) {
+
+	// We don't need anything else but the script key since we have an
+	// on-disk index for all proofs we store.
+	scriptKey := locator.ScriptKey
+	readOpts := NewAssetStoreReadTx()
+	dbErr := a.db.ExecTx(ctx, &readOpts, func(q ActiveAssetsStore) error {
+		_, err := q.FetchAssetProof(
```

Comment:
If we want this to be the more optimized version (no need to read the blob, etc), then we can have this just check the table for the primary key, instead of reading out he entire thing.

Also IIUC, this won't return true if we have the proof in the Universe tree, but not in our normal asset database, not sure if that's the intention.

Reply:
Yeah, good point. Added a `HasAssetProof` query to speed up.

> Also IIUC, this won't return true if we have the proof in the Universe tree, but not in our normal asset database, not sure if that's the intention.

Correct. The intention is to have a method for finding out which backend has the proof and which doesn't. That's why the multi archiver will return `false` if one of its backends does not have the proof, so we can import it there.